pages/lessons/projects/3.mdx

```diff
@@ -259,7 +267,11 @@ Now we need to modify the mint function with logic to be able to actually access
 }
 ```
 
-The mint function selects tiers based on the amount of native token value it receives. If it doesn't get enough for tier 0,it will give a message telling us, "Not enough value for the minimum Tier". Otherwise we can select the tier we want..... as long as we can afford it! 
+The mint function selects tiers based on the amount of native token it receives stored in `msg.value`. 
+
+The `require` statement is a simple and powerful function modifier you'll be using a lot in the future, which also helps ensure code we write is 'predictable'. In our case, if we give the function zero token value for TIER_VALUE_0, it will 'revert' with a message telling us, "Not enough value for the minimum Tier", and stop executing. Otherwise we can select the tier we want..... as long as we can afford it! 
+
+We already have two `uint256` variables declared. You might wonder which one is which inside the `tokenTier` mapping? See if you can track through the code and find out - DEFO NEED A QUESTION OR TWO ON THIS - TO GET THAT ANSWER! AND ALSO TO CHECK ON STATE AND LOCAL VARIABLES e.g. `tierId` is declared  with `0`. 
```

Comment:
Yes, this could be a really good (and difficult) question, but...
In newer versions of solidity, they incorporated naming into the mapping types, so we could avoid this confusion if we bump the solidity version in the lesson. I don't remember exactly the solidity version that started using it.

Reply:
I was going to put a side drawer in for mappings and use that to compare them to arrays, but i thought we could re-think the solidity stuff and get them all nice homogenised with the lovely content you wrote already - and get that to fit into the design that we originally talked about last year - or any updated vision of that you have.